src/main/java/org/opentripplanner/routing/impl/OrcaFareServiceImpl.java

```diff
@@ -0,0 +1,251 @@
+            case COMM_TRANS_LOCAL_SWIFT: return 1.25f;
+            case COMM_TRANS_COMMUTER_EXPRESS: return 2.00f;
+            case EVERETT_TRANSIT:
+                return fareType.equals(Fare.FareType.orcaSenior) ? 0.50f : defaultFare;
+            case PIERCE_COUNTY_TRANSIT:
+            case SEATTLE_STREET_CAR:
+            case KITSAP_TRANSIT:
+                // Pierce, Seattle Streetcar, and Kitsap only provide discounted senior fare for orca.
+                return fareType.equals(Fare.FareType.orcaSenior) ? 1.00f : defaultFare;
+            case KC_WATER_TAXI_VASHON_ISLAND: return 3.00f;
+            case KC_WATER_TAXI_WEST_SEATTLE: return 2.50f;
+            // KC Metro, Sound Transit
+            default: return 1.00f;
+        }
+    }
+
+    /**
+     * Apply Orca youth discount fares based on the ride type.
+     */
+    private float getYouthFare(RideType rideType) {
+        switch (rideType) {
+            case COMM_TRANS_LOCAL_SWIFT: return 1.75f;
+            case COMM_TRANS_COMMUTER_EXPRESS: return 3.00f;
+            case PIERCE_COUNTY_TRANSIT: return 1.00f;
+            case KC_WATER_TAXI_VASHON_ISLAND: return 4.50f;
+            case KC_WATER_TAXI_WEST_SEATTLE: return 3.75f;
+            case KITSAP_TRANSIT: return 2.00f;
+            // Default case accounts for KC_METRO, Sound Transit, Everett, and Seattle Streetcar.
+            default: return 1.50f;
+        }
+    }
+
+    /**
+     * Get the ride price for a single leg. If testing, this class is being called directly so the required agency cash
+     * values are not available therefore the default test price is used instead.
+     */
+    private float getRidePrice(Ride ride, Fare.FareType fareType, Collection<FareRuleSet> fareRules) {
+        if (IS_TEST) {
+            // Testing, return default test ride price.
+            return DEFAULT_TEST_RIDE_PRICE;
+        }
+        List<Ride> ridesSingleton = new ArrayList<>();
+        ridesSingleton.add(ride);
+        return calculateCost(fareType, ridesSingleton, fareRules);
+    }
+
+    @Override
+    protected boolean populateFare(Fare fare,
+                                   Currency currency,
+                                   Fare.FareType fareType,
+                                   List<Ride> rides,
+                                   Collection<FareRuleSet> fareRules
+    ) {
+        return calculateFare(fare, currency, fareType, rides, fareRules);
+    }
+
+    /**
+     * Calculate the cost of a journey. Where free transfers are not permitted the cash price is use. If free transfers
+     * are applicable, the most expensive discount fare across all legs is added to the final cumulative cash price.
+     */
+    public boolean calculateFare(Fare fare,
+                                 Currency currency,
+                                 Fare.FareType fareType,
+                                 List<Ride> rides,
+                                 Collection<FareRuleSet> fareRules
+    ) {
```

Comment:
Can these two methods be merged? Seems like `calculateFare` accepts the same exact arguments and is only ever called from `populateFare`.

Reply:
I've changed the access modifier for `populateFare` from `protected` to `public` and merged the two metods.